### DISCUSSION TITLE
How Scalable is Meshcentral?

### DISCUSSION THREAD

**Discussion by a-imran99** [Ylianst/MeshCentral#5825] [General]
How Scalable is meshcentral.
I have more than 7000 plus devices at the moment and I need to onboard all the devices to MeshCentral. Will the user be able to take session for more than atleast 500 different devices at the same time.
Please Suggest @Ylianst @si458 @krayon007

**Comment by si458** [comment]
i would say with 7000+ devices, i wouldnt use the nedb database, i would use a database such as mongodb

you could also look into `peering`, this allows you to setup multiple machines and peer them together so devices can connect to multiple servers and you can still control them

but from what im aware @Ylianst use to host the public meshcentral on a VM with like quad core, 8gb ram, 128gb ssd, and that handled about 5000+ devices !!!

side note: if a single user can control 500 devices at the same time, ill be amazed, and want to see proof, are u running a botnet with it? haha

**Reply by a-imran99** [reply]
@si458 haha, We have a team which tests like bots, but 500 is still out of our hands, So from your reply I understood that connection is not problem,
Say, we have 100 user who needs to take 2 remote systems at the same time which tallies to 200 remote sessions at the same time(all remote sessions are unique and no 2 users are connected to the same system)
Is this possible and runs seemlessly?

**Reply by si458** [reply]
Again in theory it should handle it. it's just basically web sockets with jpeg encoded/decoded. what might be best tho is if multiple users connect to the same computers at the same time then enable multiplexer, because in theory this will reduce the cpu load on encoding/decoding of the jpegs

**Reply by a-imran99** [reply]
got it, will check @si458
